File: bus booking pages today/tmp/fix_shadows.py
import os
import re

def fix_shadow_commas(content):
    # Regex to find shadow-[...]
    # It matches shadow-[ followed by anything that isn't ]
    pattern = r'shadow-\[([^\]]+)\]'
    
    def replace_shadow(match):
        shadow_val = match.group(1)
        # Find rgb(...) or rgba(...) inside the shadow value
        color_pattern = r'(rgba?)\(([^)]+)\)'
        
        def replace_color(color_match):
            func_name = color_match.group(1)
            args_str = color_match.group(2)
            # Split by comma or space
            args = [a.strip() for a in re.split(r'[,\s]+', args_str) if a.strip()]
            
            if len(args) == 4:
                # r, g, b, a -> r_g_b_/_a
                return f"{func_name}({args[0]}_{args[1]}_{args[2]}_/_{args[3]})"
            elif len(args) == 3:
                # r, g, b -> r_g_b
                return f"{func_name}({args[0]}_{args[1]}_{args[2]})"
            return color_match.group(0) # No change if unexpected args
            
        new_shadow_val = re.sub(color_pattern, replace_color, shadow_val)
        return f'shadow-[{new_shadow_val}]'

    return re.sub(pattern, replace_shadow, content)

def process_directory(directory):
    for root, dirs, files in os.walk(directory):
        if 'node_modules' in dirs:
            dirs.remove('node_modules')
        for file in files:
            if file.endswith(('.jsx', '.js', '.css')):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                new_content = fix_shadow_commas(content)
                
                if new_content != content:
                    with open(path, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Fixed: {path}")

if __name__ == "__main__":
    process_directory(r'c:\Users\Admin\Desktop\collab\GoAirClass\bus booking pages today\frontend\src')

File: bus booking pages today/tmp/fix_shadows.js
const fs = require('fs');
const path = require('path');

function fixShadowCommas(content) {
    // Regex for shadow-[...]
    const pattern = /shadow-\[([^\]]+)\]/g;
    
    return content.replace(pattern, (match, shadowVal) => {
        // Regex for rgb(...) or rgba(...)
        const colorPattern = /(rgba?)\(([^)]+)\)/g;
        
        const newShadowVal = shadowVal.replace(colorPattern, (colorMatch, funcName, argsStr) => {
            const args = argsStr.split(/[,\s]+/).map(a => a.trim()).filter(a => a);
            
            if (args.length === 4) {
                // r, g, b, a -> r_g_b/_a
                return `${funcName}(${args[0]}_${args[1]}_${args[2]}_/_${args[3]})`;
            } else if (args.length === 3) {
                // r, g, b -> r_g_b
                return `${funcName}(${args[0]}_${args[1]}_${args[2]})`;
            }
            return colorMatch;
        });
        
        return `shadow-[${newShadowVal}]`;
    });
}

function processDirectory(directory) {
    const files = fs.readdirSync(directory);
    
    files.forEach(file => {
        const fullPath = path.join(directory, file);
        const stats = fs.statSync(fullPath);
        
        if (stats.isDirectory()) {
            if (file !== 'node_modules') {
                processDirectory(fullPath);
            }
        } else if (file.endsWith('.jsx') || file.endsWith('.js') || file.endsWith('.css')) {
            const content = fs.readFileSync(fullPath, 'utf8');
            const newContent = fixShadowCommas(content);
            
            if (newContent !== content) {
                fs.writeFileSync(fullPath, newContent, 'utf8');
                console.log(`Fixed: ${fullPath}`);
            }
        }
    });
}

const targetDir = path.resolve('c:/Users/Admin/Desktop/collab/GoAirClass/bus booking pages today/frontend/src');
processDirectory(targetDir);
